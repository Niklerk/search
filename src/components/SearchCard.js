export const SearchCard = ({ title, snippet, url, displayUrl, engine }) => {
  return (
    <div className="card ms-3 mw-75 animate__animated animate__fadeInLeft">
      <div className="row no-gutters">
        <div className="col-md-8">
          <div className="card-body">
            <h5 className="card-title">{title}</h5>
            <p className="card-text">{snippet}</p>
            <p className="card-text">
              <small className="text-muted">{engine}</small>
            </p>
            <a className="card-link" href={url}>
              {displayUrl}
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};
