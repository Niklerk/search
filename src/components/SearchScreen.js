import { searchData } from "../data/searchData";
import { useForm } from "../hooks/useForm";
import { useDispatch, useSelector } from "react-redux";
import {
  startSearchBing,
  startSearchBoth,
  startSearchGoogle,
} from "../actions/search";
import { SearchCard } from "./SearchCard";
import Swal from "sweetalert2";
import { types } from "../types/types";

export const SearchScreen = () => {
  const dispatch = useDispatch();
  const { search } = useSelector((state) => state);
  const { loading } = useSelector((state) => state.ui);

  const [formValues, handleInputChange] = useForm({
    searchText: "",
    searchOption: "1",
  });

  const { searchText, searchOption } = formValues;

  const handleSearch = (e) => {
    e.preventDefault();
    if (searchText) {
      getResults(+searchOption);
    }
  };

  const getResults = (searchOption) => {
    switch (searchOption) {
      case 1:
        dispatch(startSearchGoogle(searchText));
        break;
      case 2:
        dispatch(startSearchBing(searchText));
        break;
      case 3:
        dispatch(startSearchBoth(searchText));
        break;
      default:
        Swal.fire("Error", types.uiErrorMessage, "error");
        break;
    }
  };
  return (
    <div className="m-3">
      <h1>Search </h1>
      <hr />
      <div>
        <form onSubmit={handleSearch} className="row g-3">
          <div className="input-group">
            <div className="col-md-6">
              <input
                autoComplete="off"
                name="searchText"
                className="form-control"
                type="text"
                placeholder="Search..."
                value={searchText}
                onChange={handleInputChange}
              />
            </div>
            <div className="col-md-3">
              <select
                name="searchOption"
                className="form-control"
                placeholder="Find your hero"
                type="text"
                onChange={handleInputChange}
                value={searchOption}
              >
                {searchData.map((data) => (
                  <option
                    className="dropdown-item"
                    key={data.id}
                    value={data.id}
                  >
                    {data.searchEngine}
                  </option>
                ))}
              </select>
            </div>
            <div className="col-md-3">
              <button
                className="btn btn-block btn-primary"
                disabled={loading}
                type="submit"
              >
                Search
              </button>
            </div>
          </div>
        </form>
      </div>
      <div className="col-12 mt-3">
        <h4>Results</h4>
        <hr />
        <div className="card-columns">
          {searchText !== "" &&
            Object.entries(search).length !== 0 &&
            search.map((item) => <SearchCard key={item.id} {...item} />)}
        </div>
      </div>
    </div>
  );
};
