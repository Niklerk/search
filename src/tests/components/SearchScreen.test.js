import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { SearchScreen } from "../../components/SearchScreen";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
  ui: {
    loading: false,
  },
};
let store = mockStore(initState);
const wrapper = mount(
  <Provider store={store}>
    <SearchScreen />
  </Provider>
);
describe("test for <SearchScreen />", () => {
  beforeEach(() => {
    store = mockStore(initState);
  });
  test("should be show the default values", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
