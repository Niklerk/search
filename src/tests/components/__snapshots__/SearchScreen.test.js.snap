// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for <SearchScreen /> should be show the default values 1`] = `
<div
  className="m-3"
>
  <h1>
    Search 
  </h1>
  <hr />
  <div>
    <form
      className="row g-3"
      onSubmit={[Function]}
    >
      <div
        className="input-group"
      >
        <div
          className="col-md-6"
        >
          <input
            autoComplete="off"
            className="form-control"
            name="searchText"
            onChange={[Function]}
            placeholder="Search..."
            type="text"
            value=""
          />
        </div>
        <div
          className="col-md-3"
        >
          <select
            className="form-control"
            name="searchOption"
            onChange={[Function]}
            placeholder="Find your hero"
            type="text"
            value="1"
          >
            <option
              className="dropdown-item"
              key="1"
              value="1"
            >
              google
            </option>
            <option
              className="dropdown-item"
              key="2"
              value="2"
            >
              bing
            </option>
            <option
              className="dropdown-item"
              key="3"
              value="3"
            >
              both
            </option>
          </select>
        </div>
        <div
          className="col-md-3"
        >
          <button
            className="btn btn-block btn-primary"
            disabled={false}
            type="submit"
          >
            Search
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    className="col-12 mt-3"
  >
    <h4>
      Results
    </h4>
    <hr />
    <div
      className="card-columns"
    />
  </div>
</div>
`;
